### ISSUE TITLE
Calls to headObject function sometimes leads to decoding errors

### ISSUE THREAD

**Issue by robertgmelo** [soto-project/soto#610]
**Summary**

As per previous discussion in Slack channel, it was requested the creation of a ticket to handle an intermittent issue when trying to access an object available on a bucket.

Here is the snippet:
```
let headObjectRequest = S3.HeadObjectRequest(bucket: bucket, key: key)
let headObject = s3.headObject(headObjectRequest)
...
```

The following error is triggered:

```
Error: dataCorrupted(Swift.DecodingError.Context(codingPath: [CodingKeys(stringValue: "x-amz-storage-class", intValue: nil)], debugDescription: "Cannot initialize StorageClass from invalid String value NONE", underlyingError: nil))
```

**How to Reproduce**

Pre-conditions:
- Have a file uploaded in a bucket

1. Try to retrieve information about the object (s3.headObject)
2. Observe the behavior

**Expected behavior**
Retrieve an information about an object should occur without problems.

**Setup (please complete the following information):**
 - OS: MacOS and Linux
 - Version of soto: latest 

**Additional context**
In fact, looking at [AWS documentation](https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadObject.html#API_HeadObject_ResponseSyntax), [`StorageClass` ](https://github.com/soto-project/soto/blob/main/Sources/Soto/Services/S3/S3_Shapes.swift#L317-L328) object is properly defined, however it seems sometimes AWS returns the **NONE** value in the header `x-amz-storage-class`.

Here is the quote from AWS documentation:

> [x-amz-storage-class](https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadObject.html#API_HeadObject_ResponseSyntax)
Provides storage class information of the object. Amazon S3 returns this header for all objects except for S3 Standard storage class objects.
For more information, see [Storage Classes](https://docs.aws.amazon.com/AmazonS3/latest/dev/storage-class-intro.html).
Valid Values: STANDARD | REDUCED_REDUNDANCY | STANDARD_IA | ONEZONE_IA | INTELLIGENT_TIERING | GLACIER | DEEP_ARCHIVE | OUTPOSTS | GLACIER_IR

**Comment by robertgmelo** [comment]
Suggestion: add `NONE` for `StorageClass`

**Comment by adam-fowler** [comment]
This should be fixed in v6.0.0
